src/support.cpp

```diff
@@ -1418,19 +1424,22 @@ void AreaSupport::generateSupportRoof(SliceDataStorage& storage, const SliceMesh
         {
             mesh_outlines.add(mesh.layers[std::round(layer_idx_above)].getOutlines());
         }
-        Polygons roofs;
-        generateSupportInterfaceLayer(global_support_areas_per_layer[layer_idx], mesh_outlines, roof_line_width, roofs);
+        Polygons roofs = generateSupportInterfaceLayer(global_support_areas_per_layer[layer_idx], mesh_outlines, roof_line_width, minimum_roof_area);
         support_layers[layer_idx].support_roof.add(roofs);
     }
 }
 
-void AreaSupport::generateSupportInterfaceLayer(Polygons& support_areas, const Polygons colliding_mesh_outlines, const coord_t safety_offset, Polygons& interface_polygons)
+Polygons AreaSupport::generateSupportInterfaceLayer(Polygons& support_areas, const Polygons colliding_mesh_outlines, const coord_t safety_offset, const double minimum_interface_area)
 {
     Polygons model = colliding_mesh_outlines.unionPolygons();
-    interface_polygons = support_areas.intersection(model);
+    Polygons interface_polygons = support_areas.intersection(model);
     interface_polygons = interface_polygons.offset(safety_offset).intersection(support_areas); //Make sure we don't generate any models that are not printable.
-    interface_polygons.removeSmallAreas(1.0);
+    if (minimum_interface_area > 0.0)
+    {
+        interface_polygons.removeSmallAreas(minimum_interface_area);
+    }
     support_areas = support_areas.difference(interface_polygons);
+    return interface_polygons;
```

Comment:
This does increase the number of copies when compared to the previous situation. Returning the polygons (without reference) means that an additional copy is done.

Reply:
Here is the place where C++ compiler is able to apply NRVO eliminating copying of the object, thus no additional overhead.